### PR TITLE
Tweak and update documentation

### PR DESCRIPTION
**Pull request by daviddrysdale**
# Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by [Cloudbuild](cloudbuild.yaml)
  - [x] I have updated [documentation](docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: docs/abi.md [hunk 1]
```diff
@@ -116,7 +116,7 @@ available on the channel, `CHANNEL_EMPTY` will be returned.
 - arg 5: Destination handle array count
 - arg 6: Address of a 4-byte location that will receive the number of handles
   retrieved with the message (as a little-endian u32)
-- return 0: Status of operation
+- return: Status of operation
```
Comment (tiziano88) [inline docs/abi.md]: Note that Wasm is getting multiple return values at some point, this was for forward compatibility with that. https://hacks.mozilla.org/2019/11/multi-value-all-the-wasm/ 

I don't mind either way anyways, if you think it's confusing.

File: docs/programming-oak.md [hunk 2]
```diff
@@ -132,11 +135,12 @@ fn main() {
 ```
 <!-- prettier-ignore-end -->
 
-The autogenerated code includes two key chunks.
+The autogenerated code includes three distinct sections.
```
Comment (tiziano88) [inline docs/programming-oak.md]: It's unclear (to me, at least) what "section" actually means here (granted, "chunks" was equally vague). Perhaps we should say that it generates three objects: a trait, a struct, and another struct, and make this a bit more structured (bullet list, or sub-headings)?

File: docs/abi.md (hunk 1 above)
Comment (daviddrysdale) [reply docs/abi.md]: Reverted (I was thinking that multiple returns are likely to far enough in the future that it's not worth allowing for them, but that link implies they might arrive sooner than I thought).

File: docs/programming-oak.md (hunk 2 above)
Comment (daviddrysdale) [reply docs/programming-oak.md]: Shifted to "parts" and added a bullet list summary at the start.

**Pull request merged**